api/clients/v2/relay_client.go

```diff
@@ -62,8 +69,8 @@ var _ RelayClient = (*relayClient)(nil)
 
 // NewRelayClient creates a new RelayClient that connects to the relays specified in the config.
 // It keeps a connection to each relay and reuses it for subsequent requests, and the connection is lazily instantiated.
-func NewRelayClient(config *RelayClientConfig, logger logging.Logger) (*relayClient, error) {
-	if config == nil || len(config.Sockets) <= 0 {
+func NewRelayClient(config *RelayClientConfig, logger logging.Logger) (RelayClient, error) {
+	if config == nil || len(config.Sockets) <= 0 || config.OperatorID == nil || config.MessageSigner == nil {
```

Comment:
should we move this check `config.OperatorID == nil || config.MessageSigner == nil` to GetChunks methods? 
Other methods like `GetBlob` should work with nil operator ID and nil signer

Reply:
I've moved these checks out of the constructor. They will cause the methods that get chunks to return an error if they are not properly configured.